miden-lib/asm/miden/contracts/wallets/basic.masm

```diff
@@ -73,3 +73,165 @@ export.send_asset.1
     # prepare the stack for return - stack has 6 elements too many
     movupw.3 dropw swap drop swap drop
 end
+
+#! Creates a note which sends the specified amount of assets out of the current account to the 
+#! specified recipient. The key difference with the `send_asset` procedure is that `send_note` 
+#! allows to create notes without assets whatsoever.
+#!
+#! Inputs stack: [tag, aux, note_type, RECIPIENT, ASSETS_HASH, ...]
+#! Advice stack: [num_assets_felt, [ASSETS]]
+#! Outputs: [note_idx, 0, 0, EMPTY_WORD, ...]
```

Comment:
Instead of assuming that asset information is on the advice stack, I think we should assume that it is in the advice map and then in this procedure we'd move it to the advice stack using the `adv.push_mapvaln` instruction.